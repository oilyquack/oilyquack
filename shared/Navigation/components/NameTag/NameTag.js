import React from "react";

import styles from "./NameTag.module.css";

const NameTag = () => (
  <div className={styles.NameTag} unselectable="on">
    <span className={styles.firstName}>
      <span>Ollie</span>
      <span>@Oily</span>
    </span>
    <span> </span>
    <span className={styles.lastName}>
      <span>Crook</span>
      <span>Quack</span>
    </span>
  </div>
);
export default NameTag;
