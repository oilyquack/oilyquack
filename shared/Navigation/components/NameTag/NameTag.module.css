@keyframes slideUpAnimation {
  0%,
  20% {
    transform: translateY(0);
  }
  30%,
  70% {
    transform: translateY(-24px);
  }
  80%,
  100% {
    transform: translateY(0);
  }
}

.NameTag {
  color: #000;
  display: flex;
  font-family: "Courier New", Courier, monospace;
  font-size: 24px;
  height: 26px;
  line-height: 24px;
  overflow: hidden;
  width: fit-content;

  -webkit-user-select: none; /* Safari */
  -moz-user-select: none; /* Firefox */
  -ms-user-select: none; /* IE10+/Edge */
  user-select: none; /* Standard */
}

.firstName {
  align-items: flex-end;
  animation: slideUpAnimation 8s infinite ease-in-out;
  display: flex;
  flex-direction: column;
}

.lastName {
  align-items: flex-start;
  animation: slideUpAnimation 8s infinite ease-in-out;
  animation-delay: 0.5s;
  display: flex;
  flex-direction: column;
}
