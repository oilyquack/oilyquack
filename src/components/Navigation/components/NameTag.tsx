import styles from "./NameTag.module.css";

const NameTag = () => (
  <div class={styles.NameTagWrapper}>
    <span class={styles.firstName}>
      <span>Ollie</span>
      <span>@Oily</span>
    </span>
    <span> </span>
    <span class={styles.lastName}>
      <span>Crook</span>
      <span>Quack</span>
    </span>
  </div>
);
export default NameTag;
