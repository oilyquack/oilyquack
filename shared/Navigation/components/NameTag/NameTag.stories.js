import React from "react";
import NameTag from "./NameTag";

export default {
  title: "NameTag",
  component: NameTag,
};

export const Primary = () => <NameTag />;

Primary.story = {
  name: "Primary",
};
