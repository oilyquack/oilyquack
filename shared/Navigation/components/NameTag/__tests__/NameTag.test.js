import { render } from "@testing-library/react";
import React from "react";

import NameTag from "../NameTag";

describe("shared : Navigation : components : NameTag", () => {
  it("renders correctly", () => {
    const { container } = render(<NameTag />);
    expect(container).toMatchSnapshot();
  });
});
