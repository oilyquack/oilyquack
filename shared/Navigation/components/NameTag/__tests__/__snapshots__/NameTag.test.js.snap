// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shared : Navigation : components : NameTag renders correctly 1`] = `
<div>
  <div
    class="NameTag"
    unselectable="on"
  >
    <span
      class="firstName"
    >
      <span>
        Ollie
      </span>
      <span>
        @Oily
      </span>
    </span>
    <span>
       
    </span>
    <span
      class="lastName"
    >
      <span>
        Crook
      </span>
      <span>
        Quack
      </span>
    </span>
  </div>
</div>
`;
